internal/provider/provider.go

```diff
@@ -161,6 +161,11 @@ func keyFromData(rootData map[string]interface{}, secretKey string) ([]byte, err
 		data = rootData
 	}
 
+	// Fail early if a the key does not exist in the secret
+	if _, ok := data[secretKey]; !ok {
+		return nil, fmt.Errorf("key %q does not exist at the secret path", secretKey)
```

Comment:
For cases where the SecretProviderClass has a larger number of secrets, it would be nice to provide a bit more detail. It will require passing more info into this function, but I think it's worth it to help users debug quicker. Maybe we should pass the whole `config.Secret` struct into this function, and include at least the object name in the error message, possibly the path as well?

Reply:
Good idea!

Instead of changing the function of the `keyFromData` (and all of its test cases). I ended up just prefixing the returned errors from this function and handle the additional detail from `GetSecret`.

This keeps the key from data decoupled from the `SecretConfig`

Reply:
Nice! That's much better